### PR TITLE
fix: Allow to run smoke test on pull request

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
Fixes building of pull-request based che-code docker image 

### What issues does this PR fix?

https://github.com/eclipse/che/issues/22341

### How to test this PR?

Just check that smoke test passed for this pull request

### CODE REVIEW

**Comment by vitaliy-guliy** [comment]
@musienko-maxim

**Review by RomanNikitenko — APPROVED** [review]

File: .github/workflows/smoke-test-pr-check.yaml [hunk 1]
```diff
@@ -71,11 +63,25 @@ jobs:
           sed -i -r -e "s|${REPLACE_FROM}||" build/dockerfiles/assembly.copy.Dockerfile
           REPLACE_FROM="COPY --from=linux-musl-content --chown=0:0 /checode-linux-musl /mnt/rootfs/checode-linux-musl"
           sed -i -r -e "s|${REPLACE_FROM}||" build/dockerfiles/assembly.copy.Dockerfile
+          
           export DOCKER_BUILDKIT=1
-          docker build -f build/dockerfiles/assembly.copy.Dockerfile -t che-code .
+          docker build --platform linux/amd64 -f build/dockerfiles/assembly.copy.Dockerfile -t che-code .
```
Comment (azatsarynnyy) [inline .github/workflows/smoke-test-pr-check.yaml]: Just a minor question, for my better understanding.
Isn't `--platform linux/amd64` redundant here and above?
Considering the `smoke-test` job is running on `ubuntu-22.04`.

**Review by azatsarynnyy — APPROVED** [review]
Thank you @vitaliy-guliy and @musienko-maxim for fixing the smoke tests PR check.
🎉

File: .github/workflows/smoke-test-pr-check.yaml (hunk 1 above)
Comment (vitaliy-guliy) [reply .github/workflows/smoke-test-pr-check.yaml]: Cleaned it up.

**Pull request merged**